converters/google/convert.go

```diff
@@ -200,7 +205,34 @@ func (c *Converter) AddResource(r TerraformResource) error {
 			// The existance of a merge function signals that this tf resource maps to a
 			// patching operation on an API resource.
 			if mapper.merge != nil {
-				converted = mapper.merge(existing.converterAsset, converted)
+				converterAsset := existing.converterAsset
+
+				// The existence of a newIamUpdater function signals that this tf
+				// resource needs to be merged with existing remote IAM data to be useful.
+				// We pre-merge that data prior to runner mapper.merge.
+				if mapper.newIamUpdater != nil && !c.offline && existing.IAMPolicy == nil {
```

Comment:
I'm imagining that eventually we will want similar functionality for other fine-grained resources like [router_nat](https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_compute_router_nat.go). 
There are several resources that operate this way, where they are essentially sub-resources that are managed individually.

If this functionality for other fine-grained resources is something we want to support in the future it may make sense to move this logic into an IAM specific block or hide it somewhere in the merge or a new fine-grained merge

Reply:
Interesting - good to know that this might have other use cases.

This solution is definitely a little hacky - I can take a stab at a slightly more factored-out approach. I had considered making it part of the mapper.merge() function - that just means that we need to pass in a converter.Config so that the merge function can make calls to the API. I wasn't sure it was worth it for just IAM but if there will be other resources that need the functionality it's probably a good idea.